homeassistant/components/here_weather/sensor.py

```diff
@@ -0,0 +1,108 @@
+"""Sensor platform for the HERE Destination Weather service."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import DOMAIN, SENSOR_TYPES
+from .utils import convert_unit_of_measurement_if_needed, get_attribute_from_here_data
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: ConfigEntry, async_add_entities
+):
+    """Add here_weather entities from a config_entry."""
+    here_weather_data_dict = hass.data[DOMAIN][config_entry.entry_id]
+
+    sensors_to_add = []
+    for sensor_type in SENSOR_TYPES:
+        for weather_attribute in SENSOR_TYPES[sensor_type]:
+            sensors_to_add.append(
+                HEREDestinationWeatherSensor(
+                    config_entry,
+                    here_weather_data_dict[sensor_type].coordinator,
+                    sensor_type,
+                    weather_attribute,
+                )
+            )
+    async_add_entities(sensors_to_add, True)
```

Comment:
The datacoordinator has updated before adding, so no need to update again.

```suggestion
    async_add_entities(sensors_to_add)
```